src/Azure/Akka.Streams.Azure.ServiceBus/ServiceBusSource.cs

```diff
@@ -28,76 +28,34 @@ public static class ServiceBusSource
-        /// </summary>
-        /// <param name="client">The client</param>
-        /// <param name="extractor">Extracts the message from the <see cref="BrokeredMessage"/></param>
-        /// <param name="maxMessageCount">The maximum number of messages to receive in a batch</param>
-        /// <param name="serverWaitTime">The time span that the server will wait for the message batch to arrive before it times out. Default = 3 seconds</param>
-        /// <param name="pollInterval">The intervall in witch the queue should be polled if it is empty. Default = 10 seconds</param>
-        public static Source<T, NotUsed> Create<T>(SubscriptionClient client, Func<BrokeredMessage, T> extractor, int maxMessageCount = 100, TimeSpan? serverWaitTime = null, TimeSpan? pollInterval = null)
-        {
-            return Source.FromGraph(new ServiceBusSource<T>(new SubscriptionClientWrapper(client), maxMessageCount, serverWaitTime, pollInterval, extractor));
+            return Source.FromGraph(new ServiceBusSource<T>(new ServiceBusClientWrapper(client), maxMessageCount, serverWaitTime, pollInterval, extractor));
         }
 
         /// <summary>
         /// Creates a <see cref="Source{TOut,TMat}"/> for the Azure ServiceBus.
         /// <para>
-        /// Note: The message of type <typeparamref name="T"/> is extracted via <see cref="BrokeredMessage.GetBody{T}"/> and the <see cref="BrokeredMessage"/> is automatically completed.
+        /// Note: The message of type <typeparamref name="T"/> is extracted via <see cref="ServiceBusReceivedMessage.Body"/> if a <see cref="messageHandler"/> is passed in, it will use that to extract the message, otherwise it will use <see cref="ServiceBusReceivedMessage.Body.ToObjectFromJson<typeparamref name="T"/>"/> and the <see cref="ServiceBusReceivedMessage"/> is automatically completed.
         /// </para>
         /// </summary>
         /// <param name="client">The client</param>
         /// <param name="maxMessageCount">The maximum number of messages to receive in a batch</param>
         /// <param name="serverWaitTime">The time span that the server will wait for the message batch to arrive before it times out. Default = 3 seconds</param>
         /// <param name="pollInterval">The intervall in witch the queue should be polled if it is empty. Default = 10 seconds</param>
-        public static Source<T, NotUsed> Create<T>(QueueClient client, int maxMessageCount = 100, TimeSpan? serverWaitTime = null, TimeSpan? pollInterval = null)
+        /// <param name="messageHandler">Custom handler to extract the message from <see cref="ServiceBusReceivedMessage.Body"/>. Default = uses <see cref="ServiceBusReceivedMessage.Body.ToObjectFromJson<typeparamref name="T"/>"/></param>
+        public static Source<T, NotUsed> Create<T>(ServiceBusReceiver client, int maxMessageCount = 100, TimeSpan? serverWaitTime = null, TimeSpan? pollInterval = null, Func<ServiceBusReceivedMessage, T> messageHandler = null)
```

Comment:
To avoid confusion, it is better to split this function into one that requires a message handler and always uses the message handler to deserialize the message and one that does not have a message handler and automatically deserialize everything using `BinaryData.ToObjectFromJson`.